spring-web/src/main/java/org/springframework/web/util/UriComponentsBuilder.java

```diff
@@ -911,7 +931,7 @@ public void append(String... pathSegments) {
 		}
 
 		@Override
-		public @Nullable PathComponent build() {
+		public @Nullable PathComponent build(boolean sanitizePath) {
```

Comment:
In the PathSegmentComponentBuilder:build method, I couldn't come up with any other approach that requires minimal changes while addressing the sanitizePath flag, especially considering that two of the three implementations make use of it. If you have a better solution, I’d be happy to implement it.